src/components/chartArea/chartArea.js

```diff
@@ -169,67 +111,169 @@ class ChartArea extends React.Component {
+      updatedChartDomain.domain = generatedDomain;
+    }
+
+    return {
+      maxY: dataSetMaxY,
+      chartDomain: { ...updatedChartDomain }
+    };
+  }
+
+  getChartLegend() {
+    const { dataSets } = this.props;
+    const legendData = [];
+
+    dataSets.forEach(dataSet => {
+      if (dataSet.legendThreshold) {
+        legendData.push({ symbol: { type: 'dash' }, ...dataSet.legendThreshold });
+      }
+
+      if (dataSet.legendData) {
+        legendData.push(dataSet.legendData);
+      }
+    });
+
+    return {
+      legendData,
+      legendOrientation: 'horizontal',
+      legendPosition: 'bottom'
+    };
+  }
+
+  render() {
+    const { chartWidth } = this.state;
+    const { dataSets, padding } = this.props;
+
+    const { isXAxisTicks, isYAxisTicks, xAxisProps, yAxisProps } = this.getChartTicks();
+    const { chartDomain, maxY } = this.getChartDomain({ isXAxisTicks, isYAxisTicks });
+    const chartLegendProps = this.getChartLegend();
+    const chartProps = { padding, ...chartLegendProps, ...chartDomain };
+
+    // FixMe: Check maxY has value, and conditionally apply ChartVoronoiContainer to avoid a massive memory leak?
+    if (maxY > 0) {
+      chartProps.containerComponent = <ChartVoronoiContainer labels={d => d.tooltip} />;
     }
 
     return (
       <div ref={this.containerRef}>
         <Chart width={chartWidth} {...chartProps}>
-          <ChartAxis {...updatedXAxisProps} />
-          <ChartAxis {...updatedYAxisProps} />
-          {(dataSetOne.thresholds && dataSetOne.thresholds.length && (
-            /** fixme: split this out into a new wrapper called ChartThreshold in PF React */
-            <ChartLine data={dataSetOne.thresholds} style={{ data: { strokeDasharray: 3.3 } }} />
-          )) ||
+          <ChartAxis {...xAxisProps} />
+          <ChartAxis {...yAxisProps} />
+          {(dataSets &&
+            dataSets.length &&
+            dataSets.map(
+              dataSet =>
+                (dataSet.thresholds && dataSet.thresholds.length && (
+                  /** fixme: split this out into a new wrapper called ChartThreshold in PF React */
+                  <ChartLine
+                    interpolation={dataSet.thresholdInterpolation || 'step'}
+                    key={helpers.generateId()}
+                    data={dataSet.thresholds}
+                    style={dataSet.thresholdStyle || { data: { strokeDasharray: 3.3 } }}
+                  />
+                )) ||
+                null
+            )) ||
             null}
           <ChartStack>
-            {(dataSetOne.data && dataSetOne.data.length && <PfChartArea data={dataSetOne.data} />) || null}
+            {(dataSets &&
+              dataSets.length &&
+              dataSets.map(
+                dataSet =>
+                  (dataSet.data && dataSet.data.length && (
+                    <PfChartArea
+                      interpolation={dataSet.dataInterpolation || 'catmullRom'}
```

Comment:
nice!

Reply:
some centripetal motion happening here... and even recursion maybe... ❤️ it.

https://en.wikipedia.org/wiki/Centripetal_Catmull%E2%80%93Rom_spline